testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -2087,6 +2087,38 @@ public void testHandlingPartitionSchemaEvolutionInPartitionMetadata()
                 ImmutableMap.of("old_partition_key", "3", "new_partition_key", "null", "value_day", "null", "value_month", "null")));
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testTrinoMetadataCompressionCodecGzip()
+    {
+        // Verify that Spark can read a table created by Trino
+        String baseTableName = "test_trino_metadata_compression_codec_gzip" + randomTableSuffix();
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+
+        onTrino().executeQuery("CREATE TABLE " + trinoTableName + "(col int) WITH (metadata_compression_codec='gzip')");
+        onTrino().executeQuery("INSERT INTO " + trinoTableName + " VALUES (1)");
+
+        assertThat(onSpark().executeQuery("SELECT * FROM " + sparkTableName)).containsOnly(row(1));
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testSparkMetadataCompressionCodecGzip()
+    {
+        // Verify that Trino can read a table created by Spark
+        String baseTableName = "test_spark_metadata_compression_codec_gzip" + randomTableSuffix();
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+
+        onSpark().executeQuery("CREATE TABLE " + sparkTableName + "(col int) USING iceberg TBLPROPERTIES ('write.metadata.compression-codec'='gzip')");
+        onSpark().executeQuery("INSERT INTO " + sparkTableName + " VALUES (1)");
+
+        assertThat(onTrino().executeQuery("SELECT * FROM " + trinoTableName)).containsOnly(row(1));
+
+        // Change 'write.metadata.compression-codec' to none and select the table in Trino
+        onSpark().executeQuery("ALTER TABLE " + sparkTableName + " SET TBLPROPERTIES ('write.metadata.compression-codec'='none')");
```

Comment:
We should probably add corresponding logic to `io.trino.plugin.iceberg.IcebergMetadata#setTableProperties` in order for this statement to be effective.